api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -34,117 +42,109 @@ public Cluster apply(org.pmiops.workbench.notebooks.model.Cluster firecloudClust
-    return ResponseEntity.ok(new EmptyResponse());
+  /**
+   * Deterministically produces a cluster name for a given user/environment.
+   * Clusters are namespaced under a free tier project, which is unique to each
+   * user, therefore the only true requirements here are to:
+   * 1. have a reproducible mapping of user -> cluster name (as we currently
+   *    don't store this)
+   * 2. generate a unique name across AoU environments (decision still pending)
+   */
+  private String clusterNameForUser() {
+      String id =
+          workbenchConfigProvider.get().server.projectId + this.userProvider.get().getEmail();
+      // Take the last 4 digits of the hash of the above.
+      return "aou-" + String.format("%04x", id.hashCode()).substring(0, 4);
   }
 
-
   @Override
-  public ResponseEntity<Cluster> getCluster(String workspaceNamespace,
-      String workspaceId) {
-
-    // This also enforces registered auth domain.
-    workspaceService.enforceWorkspaceAccessLevel(workspaceNamespace,
-        workspaceId, WorkspaceAccessLevel.WRITER);
-
-    String clusterName = this.convertClusterName(workspaceId);
-    Cluster cluster = TO_ALL_OF_US_CLUSTER.apply(
-        this.notebooksService.getCluster(workspaceNamespace, clusterName));
-    return ResponseEntity.ok(cluster);
+  public ResponseEntity<ClusterListResponse> listClusters() {
+    String project = userProvider.get().getFreeTierBillingProjectName();
+    String clusterName = this.clusterNameForUser();
+
+    org.pmiops.workbench.notebooks.model.Cluster fcCluster;
+    try {
+      fcCluster = this.notebooksService.getCluster(project, clusterName);
+    } catch (NotFoundException e) {
+      fcCluster = this.notebooksService.createCluster(project, clusterName, createFirecloudClusterRequest());
+    }
+    ClusterListResponse resp = new ClusterListResponse();
+    resp.setDefaultCluster(TO_ALL_OF_US_CLUSTER.apply(fcCluster));
+    return ResponseEntity.ok(resp);
   }
 
-
-  @Override
-  public ResponseEntity<ClusterListResponse> listClusters(String labels) {
-    List<org.pmiops.workbench.notebooks.model.Cluster> oldClusters;
-    oldClusters = this.notebooksService.listClusters(labels, /* includeDeleted */ false);
-    ClusterListResponse response = new ClusterListResponse();
-    response.setItems(oldClusters.stream().map(TO_ALL_OF_US_CLUSTER).collect(Collectors.toList()));
-    return ResponseEntity.ok(response);
+  private org.pmiops.workbench.notebooks.model.ClusterRequest createFirecloudClusterRequest() {
+    org.pmiops.workbench.notebooks.model.ClusterRequest firecloudClusterRequest = new org.pmiops.workbench.notebooks.model.ClusterRequest();
+    // TODO(calbach): Configure Jupyter server extensions here.
+    Map<String, String> labels = new HashMap<String, String>();
+    labels.put("all-of-us", "true");
+    labels.put("created-by", userProvider.get().getEmail());
+    firecloudClusterRequest.setLabels(labels);
```

Comment:
Please include this (got added on Friday):

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/api/ClusterController.java#L54